analysis/src/CompletionBackEnd.ml

```diff
@@ -1882,18 +1882,21 @@ let rec processCompletable ~debug ~full ~scope ~env ~pos ~forHover
              && (forHover || not (List.mem name identsSeen)))
       |> List.map mkLabel)
       @ keyLabels
-  | CjsxPropValue {pathToComponent; prefix; propName} -> (
+  | CjsxPropValue {pathToComponent; prefix; propName; nested} -> (
     let targetLabel =
       getJsxLabels ~componentPath:pathToComponent ~findTypeOfValue ~package
       |> List.find_opt (fun (label, _, _) -> label = propName)
     in
     let envWhereCompletionStarted = env in
     match targetLabel with
     | None -> []
-    | Some (_, typ, env) ->
-      typ
-      |> completeTypedValue ~env ~envWhereCompletionStarted ~full ~prefix
-           ~expandOption:true ~includeLocalValues:true ~completionContext:None)
+    | Some (_, typ, env) -> (
+      match typ |> resolveNested ~env ~package:full.package ~nested with
+      | None -> []
+      | Some (typ, env, completionContext) ->
+        typ
+        |> completeTypedValue ~env ~envWhereCompletionStarted ~full ~prefix
+             ~expandOption:true ~includeLocalValues:true ~completionContext))
```

Comment:
Pre-existing: not sure why `expandOption` is always `true` here. Should there be a check that the label is optional?

Reply:
I'd love to, but I don't know how to figure that out in JSX. I'd need to parse the actual component and look whether the labelled arg in the make function is optional, right?

Reply:
Yes, this can be considered later I guess.